pims/base_frames.py

```diff
@@ -348,11 +479,18 @@ class FramesSequenceND(FramesSequence):
     >>> frames.default_coords['m'] = 3
     >>> frames[5]  # returns Frame at T=5, M=3 with shape (2, 10, 64, 64)
     """
+    def _register_get_frame(self, method, axes):
+        axes = tuple([a for a in axes])
+        if not hasattr(self, '_get_frame_dict'):
+            self._get_frame_dict = dict()
```

Comment:
I would suggest putting this initialization in `FrameSequenceND.__init__` and requrining sub-classes to call `super()` before doing anything else.   Likely will have to do this test + warn for awhile to maintain back-compat.